libs/utils/src/lock_file.rs

```diff
@@ -0,0 +1,77 @@
+//! A module to allow processes to lock on specific FS file exclusive.
+//! Could be used to forbid multiple processes to start in the same directory.
+//!
+//! Note that locked files are done via [`fcntl::flock`] which means that
+//! * files are locked only for taking another lock on them: all other operations are still possible on files.
+//! For example, other process could do `rm file` and that would be allowed.
+//! * if no file exists (after `rm` in another process), the process that "locked" the file still runs and does not error
+//! new procesess are able to start in the same directory, as they will be creating another file to lock.
+
+use std::{
+    fs,
+    os::unix::prelude::{AsRawFd, OpenOptionsExt},
+    path::Path,
+};
+
+use anyhow::Context;
+use nix::{fcntl, libc};
+
+use crate::crashsafe;
+
+pub enum LockCreationResult {
+    Created {
+        new_lock_contents: String,
+        file: fs::File,
+    },
+    AlreadyLocked {
+        existing_lock_contents: String,
+    },
+    CreationFailed(anyhow::Error),
+}
+
+/// Creates a lock file in the path given and writes the given contents into the file.
+pub fn create_lock_file<F>(lock_file: &Path, contents_producer: F) -> LockCreationResult
```

Comment:
All the callers just pass a short string, so the `contents_producer` callback seems overly fancy here. Let's make it just `content: &str`.